comp/otelcol/otlp/components/exporter/datadogexporter/config.go

```diff
@@ -0,0 +1,437 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+package datadogexporter
+
+import (
+	"encoding"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/exporter/serializerexporter"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname/validate"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/config/configopaque"
+	"go.opentelemetry.io/collector/config/configretry"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.uber.org/zap"
+)
+
+var (
+	errUnsetAPIKey   = errors.New("api.key is not set")
+	errNoMetadata    = errors.New("only_metadata can't be enabled when host_metadata::enabled = false or host_metadata::hostname_source != first_resource")
+	errEmptyEndpoint = errors.New("endpoint cannot be empty")
+)
+
+const (
+	// DefaultSite is the default site of the Datadog intake to send data to
+	DefaultSite = "datadoghq.com"
+)
+
+// APIConfig defines the API configuration options
+type APIConfig struct {
+	// Key is the Datadog API key to associate your Agent's data with your organization.
```

Comment:
This config is same as Datadog exporter config. In future we should ensure both exporters use same config.